docs/javascript.md

```diff
@@ -127,6 +127,22 @@ const number = Generators.input(numberInput);
 
 Expressions cannot declare top-level reactive variables. To declare a variable, use a code block instead. You can declare a variable in a code block (without displaying it) and then display it somewhere else using an inline expression.
 
+## TypeScript <a href="https://github.com/observablehq/framework/pull/1632" class="observablehq-version-badge" data-version="prerelease" title="Added in #1632"></a>
+
+TypeScript fenced code blocks (<code>```ts</code>) allow TypeScript to be used in place of JavaScript. You can also import TypeScript modules (`.ts`). Use the `.js` file extension when importing TypeScript modules; TypeScript is transpiled to JavaScript during build.
+
+<div class="warning">
+
+Framework does not perform type checking during preview or build. If you want the additional safety of type checks, considering using [`tsc`](https://www.typescriptlang.org/docs/handbook/compiler-options.html).
+
+</div>
+
+<div class="note">
+
+TypeScript fenced code blocks do not currently support [implicit display](#implicit-display), and TypeScript is not currently allowed in [inline expressions](#inline-expressions).
+
+</div>
+
```

Comment:
```suggestion
```

I don't feel this is strictly necessary, as the documentation for implicit display starts with "JavaScript". We could add "JavaScript" before inline expressions, too.

Otherwise, "currently" seems to hint that these might be future goals. I'm not sure it's necessarily the case; if it is and if we want to keep this note, we could link to an issue to see if there is demand?

Reply:
I think it is important to keep because it’s natural for people to otherwise assume that there is parity/consistency between TypeScript and JavaScript when that is not the case. I want to highlight the ways they are different.

We could remove the word “currently”, but I was curious to see if people would ask for it. If we don’t include that word, it may imply that we reject future support for it, which might discourage people from asking for it — and then we wouldn’t get a sense if there’s demand. I think we’re open to adding in the future (at least if it’s easy).